spyder_kernels/comms/frontendcomm.py

```diff
@@ -27,8 +43,81 @@ def __init__(self, kernel):
         self.kernel.comm_manager.register_target(
             self._comm_name, self._comm_open)
 
+        self.comm_port = None
+
+        if self.kernel.parent:
+            # self.kernel.parent is IPKernelApp unless we are in tests
+            # Create a new socket
+            context = zmq.Context()
+            self.comm_socket = context.socket(zmq.ROUTER)
+            self.comm_socket.linger = 1000
+
+            self.comm_port = get_port()
+
+            self.comm_port = self.kernel.parent._bind_socket(
+                self.comm_socket, self.comm_port)
+            if hasattr(zmq, 'ROUTER_HANDOVER'):
+                # set router-handover to workaround zeromq reconnect problems
+                # in certain rare circumstances
+                # see ipython/ipykernel#270 and zeromq/libzmq#2892
+                self.comm_socket.router_handover = 1
+
+            self.comm_thread_close = threading.Event()
+            self.comm_socket_thread = threading.Thread(target=self.poll_thread)
+            self.comm_socket_thread.start()
+
+            # patch parent.close
+            if not PY2:
+                super_close = self.kernel.parent.close
+
+                def close():
+                    """Close comm_socket_thread."""
+                    self.comm_thread_close.set()
+                    context.term()
+                    self.comm_socket_thread.join()
+                    super_close()
+
+                self.kernel.parent.close = close
+
+    def poll_thread(self):
+        """Recieve messages from comm socket."""
         if not PY2:
-            self._main_thread_id = threading.get_ident()
+            # Create an event loop to handle some messages
```

Comment:
Let's remove `some` in this comment, because I guess this event loop handles all messages, right?

Reply:
It is only needed if `out_stream` is used by the handler. For now this is only used by the complete requests.